### PR TITLE
refactor(app): add collapsible sections and card functionality to devices landing page

### LINKED ISSUE #8674: Devices Landing: "Why is my robot not available?" modal

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

- Add collapsible "available" and "unavailable" sections to the device landing page
- Add "setting up a new robot" help modal
- Device card link from whole card click, styling tweaks

Closes #8674

# Review requests

- navigate to Devices landing page
- if present, all available robots should appear in collapsible "Available Section" with proper count
- if present, all unavailable robot should appear in collapsible "Unavailable Section" with proper count
- "setting up a new robot" link opens a help modal, which contains a live link to a support article continuing the explanation. 

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/Devices/RobotCard.tsx [hunk 1]
```diff
@@ -5,27 +5,25 @@ import { Link } from 'react-router-dom'
...
-  Text,
   ALIGN_CENTER,
   ALIGN_START,
-  C_BLUE,
   C_MED_LIGHT_GRAY,
   C_WHITE,
   DIRECTION_COLUMN,
   DIRECTION_ROW,
-  SIZE_2,
   SPACING_2,
```
Comment (koji) [inline app/src/organisms/Devices/RobotCard.tsx]: Using SPACING would be better.
SPACING_2 ->  SPACING.spacing3
SPACING_3 ->  SPACING.spacing4

File: app/src/atoms/Slideout/index.tsx [hunk 2]
```diff
@@ -36,7 +36,11 @@ const EXPANDED_STYLE = css`
   overflow-x: hidden;
   width: 19.5rem;
   max-width: 19.5rem;
+<<<<<<< Updated upstream
   height: 100%;
+=======
+  visibility: visible;
+>>>>>>> Stashed changes
```
Comment (koji) [inline app/src/atoms/Slideout/index.tsx]: line 39, 41, and 43 are unnecessary, right?

File: app/src/pages/Devices/DevicesLanding/__tests__/NewRobotSetupHelp.test.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import * as React from 'react'
+import { renderWithProviders } from '@opentrons/components'
+
+import { i18n } from '../../../../i18n'
+import { NewRobotSetupHelp } from '../NewRobotSetupHelp'
+import { fireEvent } from '@testing-library/react'
```
Comment (koji) [inline app/src/pages/Devices/DevicesLanding/__tests__/NewRobotSetupHelp.test.tsx]: probably between `react` and `@opentrons/components`

File: app/src/pages/Devices/DevicesLanding/index.tsx [hunk 4]
```diff
@@ -5,67 +5,73 @@ import { useSelector } from 'react-redux'
...
-import {
-  getConnectableRobots,
-  getReachableRobots,
-  getUnreachableRobots,
-  getScanning,
-} from '../../../redux/discovery'
+import { CollapsibleSection } from '../../../molecules/CollapsibleSection'
+import { getScanning } from '../../../redux/discovery'
 
 import type { State } from '../../../redux/types'
```
Comment (koji) [inline app/src/pages/Devices/DevicesLanding/index.tsx]: This line should be after `import { NewRobotSetupHelp } from './NewRobotSetupHelp'`

**Review by koji — APPROVED** [review]
A few small changes would be needed but otherwise LGTM!

File: app/src/atoms/Slideout/index.tsx (hunk 2 above)
Comment (b-cooper) [reply app/src/atoms/Slideout/index.tsx]: whoops, good call, I'm not sure how this conflict snuck in.

**Pull request merged**